scripts/lib/CIME/case/case.py

```diff
@@ -1123,6 +1128,23 @@ def configure(self, compset_name, grid_name, machine_name=None,
         if test:
             self.set_value("TEST",True)
 
+        # sanity check: when use pgi-gpu and nvhpc-gpu compiler, we must have
+        #               at least one gpu per node available
+        if compiler in ["pgi-gpu", "nvhpc-gpu"]:
```

Comment:
Consider changing this to be 
if "gpu" in compiler:

Reply:
Thanks @jedwards4b. Will update it soon!